tests/testthat/test-PCA.R

```diff
@@ -0,0 +1,109 @@
+                 prob = c(0.18, 0.1, 0.1, 0.6, 0.01, 0.01))
+)
+df$char_DTS <- as.character(df$char_DTS)
+# give is_ween likeliness score
+df["is_ween"] <- df["length"] - 1 * df["weirdness"] + 2
+df$is_ween[df["genre"] == "Rock"]  <-
+  df$is_ween[df["genre"] == "Rock"] + 2
+df$is_ween[df["genre"] == "Jazz"]  <-
+  df$is_ween[df["genre"] == "Jazz"] - 1
+df$is_ween[df["genre"] == "Country"]  <-
+  df$is_ween[df["genre"] == "Country"] - 1
+df$is_ween[df["reaction"] == "Huh"] <-
+  df$is_ween[df["reaction"] == "Huh"] + 1
+df$is_ween[df["reaction"] == "Love"] <-
+  df$is_ween[df["reaction"] == "Love"] + 2
+df$is_ween[df["reaction"] == "Mixed"] <-
+  df$is_ween[df["reaction"] == "Mixed"] - 1
+df$is_ween[df["reaction"] == "Dislike"] <-
+  df$is_ween[df["reaction"] == "Dislike"] - 4
+
+# Add missing data
+df$reaction[sample(1:n, 32, replace = FALSE)] <- NA
+df$length[sample(1:n, 51, replace = FALSE)] <- NA
+df$genre[sample(1:n, 125, replace = FALSE)] <- NA
+df$weirdness[sample(1:n, 9, replace = FALSE)] <- NA
+
+train_index <- caret::createDataPartition(
+  df$is_ween,
+  p = 0.8,
+  times = 1,
+  list = TRUE)
+
+d_train <- df[train_index$Resample1, ]
+d_test <- df[-train_index$Resample1, ]
+
+d_train$length[1] <- d_test$length[1] <- NA
+d_train$reaction[2] <- d_test$reaction[2] <- NA
+d_train$weirdness[3] <-  d_test$weirdness[3] <- NA
+d_train$genre[3] <- d_test$genre[3] <- NA
+
+
+
+
+
+#TEST - Make sure error with missing data
+test_that("Can't run with missing data", {
+  expect_error(prep_data(d = d_train, outcome = is_ween, impute = FALSE, PCA = 5))
+})
+
+#TEST - Make sure number of PC columns matches input param
+test_that("Number of PC columns matches input param", {
+  nPCs <- 7
+  d_PCA <- prep_data(d = d_train, outcome = is_ween, PCA = nPCs)
+  expect_equal(d_PCA[, grepl("PC", names(d_PCA))] %>% ncol(), nPCs)
+})
+
+#PCA ignores ignored columns and outcome variables
+test_that("PCA ignores outcome and ignored cols", {
+  d_PCAignored <- prep_data(d = d_train, song_id, outcome = is_ween, PCA = 5)
+  d_train2 <- d_train[, !(names(d_train) %in% c("song_id", "is_ween"))]
+  expect_equal(ncol(d_train2[, names(d_train2) %in% names(d_PCAignored)]), 0)
+  expect_equal(ncol(d_PCAignored[, names(d_PCAignored) %in% c("song_id", "is_ween")]), 2)
+})
+
+#TEST - Make sure errors if too high param
+test_that("Errors if number of PCs is higher than width of prepped dataframe", {
+  dfwidth <- ncol(prep_data(d = d_train, outcome = is_ween))
+  expect_error(prep_data(d = d_train, outcome = is_ween, PCA = (dfwidth + 1)))
+})
+
+#TEST - Make sure flash_models, tune_models, and predict work for pca'd data
+test_that("flash_models, tune_models, and predict work", {
+  d_prep <- prep_data(d = d_train, song_id, outcome = is_ween, PCA = 5)
+  expect_error(pca_mod <- tune_models(d_prep, outcome = is_ween, tune_depth = 1), NA)
+  expect_error(pca_flash <- flash_models(d_prep, outcome = is_ween), NA)
+  expect_error(pca_oldpreds <- predict(pca_mod), NA)
```

Comment:
add another set of tests here to make sure the number of columns out of predict is correct and that there are no NAs in them.